addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java

```diff
@@ -206,23 +210,29 @@ private AutomationProgress runPlan(LinkedHashMap<?, ?> envData, ArrayList<?> job
                             Constant.messages.getString("automation.error.job.data", paramsObj));
                     continue;
                 }
-                job.applyParameters((LinkedHashMap<?, ?>) paramsObj, progress);
-
-                // In case errors or warnings are encountered in the applyParameters() call above
-                if (env.isTimeToQuit()) {
-                    break;
-                }
-
-                progress.info(
-                        Constant.messages.getString("automation.info.jobstart", job.getType()));
-                job.setEnv(env);
-                job.runJob(env, jobData, progress);
-                progress.info(Constant.messages.getString("automation.info.jobend", job.getType()));
+                job.verifyParameters((LinkedHashMap<?, ?>) paramsObj, progress);
+                job.verifyJobSpecificData(jobData, progress);
+                jobsToRun.put(job, jobData);
             } else {
                 progress.error(
                         Constant.messages.getString("automation.error.job.unknown", jobType));
             }
         }
+
+        if (env.isTimeToQuit()) {
+            return progress;
+        }
+
+        for (Entry<AutomationJob, LinkedHashMap<?, ?>> jobInfo : jobsToRun.entrySet()) {
```

Comment:
We should still check `env.isTimeToQuit()` in between each job as there could be runtime errors. For example is a target app is unavailable then the config could still be valid but any job that tries to access ot will fail.